[16.0][ADD] sale_quotation_customer_quotation_workflow_notifications: notifications handeling for "CQ" workflow.

nicolas-delbovier-acsone opened the pull request:
This PR is based on #1597

* New "Customer Quotation" message subtypes that we can subscribe partners to
* Ensures salesman receives a notification for customer quotation actions
* Returns a confirmation notification to  a customer requesting a quotation
* Notifies salesman when a user requests a quotation
* Toggles between "CQ" workflow default message types and base default message types when toggling the `use_customer_quotation_workflow` variable.

<img width="1189" height="469" alt="image" src="https://github.com/user-attachments/assets/aa615c70-9a6d-49bc-b4f0-960324c7c63e" />
<img width="1189" height="469" alt="image" src="https://github.com/user-attachments/assets/48ea12bb-75f0-4dd7-9a04-d0ebea625995" />

nicolas-delbovier-acsone committed:
[ADD] alc_sale_quotation_customer_quotation_workflow_notifications: notifications handeling for "CQ" workflow.

* New "Customer Quotation" message subtypes that we can subscribe partners to
* Ensures salesman receives a notification for customer quotation actions
* Returns a confirmation notification to  a customer requesting a quotation
* Notifies salesman when a user requests a quotation
* Toggles between "CQ" workflow default message types and base default message types when toggling the `use_customer_quotation_workflow` variable.

sebastienbeau commented:
Interesting feature, I will take the time to test it and review it. @nicolas-delbovier-acsone do you plan to be in OCA days next week ?

nicolas-delbovier-acsone commented:
> Interesting feature, I will take the time to test it and review it. @nicolas-delbovier-acsone do you plan to be in OCA days next week ?

Yes I will be there, I would be happy to discuss this further in person then.

sebastienbeau commented:
Cool ! Let's discuss about this on Monday ;)

nicolas-delbovier-acsone committed:
[IMP] sale_quotation_customer_quotation_workflow_notifications: fix line too long

nicolas-delbovier-acsone committed:
[IMP] sale_quotation_customer_quotation_workflow_notifications: simplify code.

This commit removes a redundant check for `sale_order.use_customer_quotation_workflow` to simplify the code logic.
